### ISSUE TITLE
No longer runs in node 0.10.x

### ISSUE THREAD

**Issue by brakon** [BryanDonovan/node-cache-manager#46]
Updating from cache-manager 1.1.0 to 1.4.0 does not run on Node 0.10.x

Unhandled rejection ReferenceError: Promise is not defined
    at Object.self.set (/Users/bbergendahl/projects/bc-frontend/node_modules/cache-manager/lib/stores/memory.js:31:20)

**Comment by BryanDonovan** [comment]
@brakon - thanks for the report.  I've pushed a feature branch called `feature/non-promise-memory-store` that hopefully addresses this for you.  You'll have to change your code to pass in a `noPromises` option like this though:

```javascript
var cache = caching({store: memoryStore.create({noPromises: true})});
```
Would that work?

An alternative would be to create your own custom store that does what you need (which is really the intent of this library -- the built-in memory store probably won't be right for everyone).

**Comment by BryanDonovan** [comment]
p.s., you can try out this branch by using this in your package.json:

```json
 "cache-manager": "git+https://github.com/BryanDonovan/node-cache-manager.git#feature/non-promise-memory-store"
```

**Comment by PuKoren** [comment]
I think this is because set expect the callback function to be defined and fallback to promise mode if not. You fix is good but requires a configuration change, we could check if Promise is defined before using it, in case where callback is not defined, what do you think? (this way we can have a backward compatible fix, since the add of Promises should not have been a breaking change - my bad)

**Comment by BryanDonovan** [comment]
@PuKoren - yes, that would be good.  Thanks!

**Comment by PuKoren** [comment]
@BryanDonovan not sure if node will like to check for undefined variable, I think ``if(Promise)`` will crash :/

Maybe ``typeof Promise === 'undefined'`` ?

**Comment by BryanDonovan** [comment]
Yeah, I think that will work.

**Comment by BryanDonovan** [comment]
This is fixed in version 1.4.1
